src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs

```diff
@@ -1203,7 +1206,26 @@ private async ValueTask<Http2Stream> SendHeadersAsync(HttpRequestMessage request
             // in order to avoid consuming resources in potentially many requests waiting for access.
             try
             {
-                await _concurrentStreams.RequestCreditAsync(1, cancellationToken).ConfigureAwait(false);
+                if (_pool.EnableMultipleHttp2Connections)
+                {
+                    if (!_concurrentStreams.TryRequestCreditNoWait(1))
+                    {
+                        // Maximum number of streams reached
+                        _canAddNewStream = false;
+                        if (_pool.ThrowOnStreamLimitReached)
+                        {
+                            throw new HttpRequestException(null, null, RequestRetryType.RetryOnSameOrNextProxy);
```

Comment:
Not the new RetryOnNextConnection?

RetryOnNextConnection never seems to be thrown.

Reply:
It got lost during refactoring. Fixed.